lib/Core/GeoJson.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  feature,
+  Feature,
+  featureCollection,
+  FeatureCollection,
+  Geometries,
+  Geometry,
+  GeometryCollection,
+  LineString,
+  MultiLineString,
+  MultiPoint,
+  MultiPolygon,
+  Point,
+  Polygon,
+  Properties
+} from "@turf/helpers";
+import { isJsonObject, isJsonString } from "./Json";
+
+export type GeoJsonCrs =
+  | {
+      type: "name";
+      properties: {
+        name: string;
+      };
+    }
+  | {
+      type: "EPSG";
+      properties: {
+        code: string;
+      };
+    };
+
+export type FeatureCollectionWithCrs<
+  G = Geometry | GeometryCollection,
+  P = Properties
+> = FeatureCollection<G, P> & {
+  crs?: GeoJsonCrs;
+};
+// Note: these type checks are not that rigorous, we are assuming we are getting valid GeoJson objects
+export function isFeatureCollection(
+  json: any
+): json is FeatureCollectionWithCrs {
+  return (
+    isJsonObject(json, false) &&
+    json.type === "FeatureCollection" &&
+    Array.isArray(json.features)
+  );
+}
+
+export function isFeature(json: any): json is Feature {
+  return (
+    isJsonObject(json, false) && json.type === "Feature" && !!json.geometry
+  );
+}
+
+export function isPoint(json: any): json is Feature<Point> {
+  return (
+    isJsonObject(json, false) &&
+    json.type === "Feature" &&
+    isJsonObject(json.geometry, false) &&
```

Comment:
These 3 lines are repeated in almost every function, is there a name for what they are checking so that check could be factored out to its own function?